Add the ability to remind users to register to vote via iCalendar events

krismuniz opened the pull request:
## Description

Añade la capacidad de añadir tres recordatorios para inscribirse en una Junta de Inscripción Permanente.

Crea dos archivos tipo `.ics` (uno en inglés y el otro en español). Cada file enumera tres iCalendar events:

- 1 de septiembre de 2020: Faltan dos semanas! [...]
- 7 de septiembre de 2020: Falta una semana! [...]
- 14 de septiembre de 2020: Hoy es el último día para [...]

(parafraseado)

## Additional Context

https://github.com/Code4PuertoRico/paravotar/issues/44#issuecomment-669649108 Este PR probablemente esté bloqueado hasta que nos pongamos de acuerdo en cuanto al "copy" que debemos usar para el site y para los eventos del calendario. 

Closes #44

## Next Steps

Probablemente sea necesario generalizar la manera en que se generan los iCalendar files para otros recordatorios a lo largo de la vida del proyecto. Por ejemplo: añadir un recordatorio para la fecha de las primarias y/o elecciones generales. Incluso darle al usuario la habilidad de escoger cuál es su junta de inscripción permanente y añadir la dirección a la invitación.

Hay Gatsby plugins que podrían ayudar con esto. También ver esta discusión: https://github.com/Code4PuertoRico/paravotar/issues/44#issuecomment-669646093

## Screenshots

### Desktop

<img width="1286" alt="Screen Shot 2020-08-06 at 10 07 33 PM" src="https://user-images.githubusercontent.com/860507/89611254-8e436500-d831-11ea-9940-88ea3cd18dca.png">
<img width="1286" alt="Screen Shot 2020-08-06 at 10 07 26 PM" src="https://user-images.githubusercontent.com/860507/89611258-91d6ec00-d831-11ea-9c60-3b9cacf7d957.png">

### Mobile (iOS)

![IMG_2644](https://user-images.githubusercontent.com/860507/89611300-b03ce780-d831-11ea-97e3-0093cd7c09c4.PNG)

### Gracias!

krismuniz committed:
Add English and Spanish iCalendar files

Three events per .ics file:

- Sept 1st
- Sept 7th
- Sept 14th

krismuniz committed:
Add reminder-illustration.png

Compressed via tinypng.com

krismuniz committed:
Add support for download links for Link component

- treat download links as regular non-Gatsby <a> elements

krismuniz committed:
Add support for enrollment reminders

- introduces four new localization strings:
  - add-to-calendar
  - enrollment-reminder
  - enrollment-reminder-guide1
  - enrollment-reminder-guide2
- renders reminder-illustration.png

krismuniz committed:
Restore date-fns dependency

krismuniz committed:
Enhance flexbox layout on larger screens

- reverse row order
- vertically-align items center

froi commented:
This looks sooooooo cool @krismuniz.

Seem to be a dependency issues. @eluciano11 @Jpadilla1 

```
warning Browserslist: caniuse-lite is outdated. Please run the following command: `yarn upgrade`
  
   ⚠️  Tailwind is not purging unused styles because no template paths have been provided.
      If you have manually configured PurgeCSS outside of Tailwind or are deliberately not
      removing unused styles, set `purge: false` in your Tailwind config file to silence
      this warning.
   
      https://tailwindcss.com/docs/controlling-file-size/#removing-unused-css
  
   ⚠️  Tailwind is not purging unused styles because no template paths have been provided.
      If you have manually configured PurgeCSS outside of Tailwind or are deliberately not
      removing unused styles, set `purge: false` in your Tailwind config file to silence
      this warning.
   
      https://tailwindcss.com/docs/controlling-file-size/#removing-unused-css
failed Building production JavaScript and CSS bundles - 47.331s
error Generating JavaScript bundles failed
```

krismuniz commented:
Local error output from `yarn build` is the following:

```
failed Building production JavaScript and CSS bundles - 35.623s

ERROR #98123  WEBPACK

Generating JavaScript bundles failed

Bundle Analyzer - Token not found, please specify a token using BUNDLE_ANALYZER_TOKEN env variable

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Looking at the `build.yml` workflow, there's a reference to `BUNDLE_ANALYZER_TOKEN`. It appears that since my branch is in a fork of the `code4puertorico/paravotar` repo, `{{ secrets.BUNDLE_ANALYZER_TOKEN }}` is undefined.

eluciano11 commented on src/locales/es/translations.json:
```diff
@@ -36,6 +36,10 @@
+    "add-to-calendar": "Añadir al Calendario",
+    "enrollment-reminder": "Programa tu visita para que no pierdas tu derecho al voto",
+    "enrollment-reminder-guide1": "Crea un recordatorio automático para que no olvides inscribirte",
+    "enrollment-reminder-guide2": "Añade tres eventos a tu calendario para recordarte hacer la gestión de sacar tu tarjeta electoral antes de la fecha límite para inscribirse. No lo dejes para lo último. Te damos 3 chances para que lo logres. Vamos a ti!",
```
```suggestion
    "enrollment-reminder-guide2": "Añade tres eventos a tu calendario para recordarte hacer la gestión de sacar tu tarjeta electoral antes de la fecha límite para inscribirse. No lo dejes para lo último. Te damos 3 chances para que lo logres. ¡Vamos a ti!",
```

eluciano11 commented:
@froi eso del warning que mencionas lo tenemos que arreglar pero creo que el issue que estamos viendo es por lo que menciona @krismuniz.

@krismuniz awesome work, todo se ve super bien! tengo un minor feedback que debemos mirar antes de merge:

1. Debemos añadir esta sección al sidebar y al navbar de mobile. Yo le pondría como título "Crea un recordatorio" o algo así.
<img width="1594" alt="Screen_Shot_2020-08-08_at_10_57_10_AM" src="https://user-images.githubusercontent.com/6109151/89713677-563e3e00-d967-11ea-95b9-a6141a92bc15.png">

2. Cuando intenté abrir el ics en mi Mac no me funciono. No creo que esto sea un big issue pq la mayoría del trafico es de celulares pero creo que debemos verificar qué esta pasando.
![image](https://user-images.githubusercontent.com/6109151/89713552-6c97ca00-d966-11ea-8b72-c61a1e84076d.png)

froi commented:
@krismuniz

> Looking at the build.yml workflow, there's a reference to BUNDLE_ANALYZER_TOKEN. It appears that since my branch is in a fork of the code4puertorico/paravotar repo, {{ secrets.BUNDLE_ANALYZER_TOKEN }} is undefined.

Hmmmm, strange. It shouldn’t matter if you’re pushing a fork. The PR, the checks, and the Action should be running in this repo’s context not yours. I’ll take a 👀.

krismuniz committed:
Fix invalid .ics files

- wrap lines at no more than 75 characters
- remove empty lines
- add description to VALARM
- escape potentially-conflicting special characters

krismuniz committed:
Fix incorrect id for heading

krismuniz committed:
Update yarn dependencies

krismuniz committed:
Add enrollment reminder link to navigation components

- introduces one new localization string:
  - nav.enrollment-reminder

krismuniz commented:
Ya hice los cambios necesarios:

- añadí el símbolo de exclamación que olvidé
- arreglé los iCalendar files (los pasé por un validation tool y no estaban up to standard AT ALL 😂)
- arreglé un incorrect id en el heading del componente `<EnrollmentReminder />`
- añadí navigation links al sidebar y mobile navbar para `Crea un Recordatorio`

Aprendí un par de cosas importantes de iCalendar files:

1. Lines should be hard-wrapped at 75 characters to conform to the standard
2. Empty lines are not specified so some calendar clients don't know what to do with them
3. Some characters like commas need to be escaped :)
4. You shouldn't try to be clever and edit them manually unless you have a lot of patience and know what you are doing
5. This validator is super handy: https://icalendar.org/validator.html

@eluciano11

krismuniz committed:
Temporarily disable the Bundle Analyzer Gatsby plugin

krismuniz committed:
Merge github.com:Code4PuertoRico/paravotar into enrollment-reminders

# Conflicts:
#	src/components/navbar.tsx
#	src/components/sidebar.tsx
#	src/locales/en/translations.json
#	src/locales/es/translations.json
#	src/pages/index.tsx
#	yarn.lock

krismuniz committed:
Move `.ics` files from `src/assets/ics` to `static/ics`

krismuniz committed:
Conditionally include @bundle-analyzer/gatsby-plugin plugin

- ignore if process.env.BUNDLE_ANALYZER_TOKEN does not exist

eluciano11 commented:
🚀 

@krismuniz Amazing work, gracias por todo el trabajo, por compartir la info de los ics y el ics validator!

💪 👏
